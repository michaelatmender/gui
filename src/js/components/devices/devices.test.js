import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { createMount } from '@material-ui/core/test-utils';
import Devices from './devices';

it('renders correctly', () => {
  const context = {
    childContextTypes: {
      router: () => {}
    },
    context: {
      router: {
        route: {
          location: {
            hash: '',
            pathname: '',
            search: '',
            state: ''
          },
          match: { params: {}, isExact: false, path: '', url: '' }
        }
      }
    }
  };
  const tree = createMount()(
    <MemoryRouter>
      <Devices />
    </MemoryRouter>,
    context
  );
  expect(tree.html()).toMatchSnapshot();
});
