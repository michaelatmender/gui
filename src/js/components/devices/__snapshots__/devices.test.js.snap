// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `"<div><div class=\\"MuiTabs-root\\"><div class=\\"MuiTabs-scroller MuiTabs-fixed\\" style=\\"overflow: hidden;\\"><div class=\\"MuiTabs-flexContainer\\" role=\\"tablist\\"><a class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit Mui-selected\\" tabindex=\\"0\\" role=\\"tab\\" aria-disabled=\\"false\\" aria-selected=\\"true\\" href=\\"/devices\\"><span class=\\"MuiTab-wrapper\\">Device groups</span><span class=\\"MuiTouchRipple-root\\"></span></a><a class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit\\" tabindex=\\"0\\" role=\\"tab\\" aria-disabled=\\"false\\" aria-selected=\\"false\\" href=\\"/devices/pending\\"><span class=\\"MuiTab-wrapper\\">Pending</span><span class=\\"MuiTouchRipple-root\\"></span></a><a class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit\\" tabindex=\\"0\\" role=\\"tab\\" aria-disabled=\\"false\\" aria-selected=\\"false\\" href=\\"/devices/preauthorized\\"><span class=\\"MuiTab-wrapper\\">Preauthorized</span><span class=\\"MuiTouchRipple-root\\"></span></a><a class=\\"MuiButtonBase-root MuiTab-root MuiTab-textColorInherit\\" tabindex=\\"0\\" role=\\"tab\\" aria-disabled=\\"false\\" aria-selected=\\"false\\" href=\\"/devices/rejected\\"><span class=\\"MuiTab-wrapper\\">Rejected</span><span class=\\"MuiTouchRipple-root\\"></span></a></div><span class=\\"PrivateTabIndicator-root-25 PrivateTabIndicator-colorSecondary-27 MuiTabs-indicator\\" style=\\"left: 0px; width: 0px;\\"></span></div></div><div class=\\"tab-container\\"><div class=\\"leftFixed\\"><div><div class=\\"muted margin-bottom-small\\">Groups</div><ul class=\\"MuiList-root MuiList-padding\\"><div class=\\"MuiButtonBase-root MuiListItem-root grouplist MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" style=\\"background-color: rgb(231, 231, 231);\\"><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\"><span>All devices</span></span></div><span class=\\"MuiTouchRipple-root\\"></span></div><hr class=\\"MuiDivider-root\\"><div class=\\"MuiButtonBase-root MuiListItem-root grouplist MuiListItem-gutters Mui-disabled MuiListItem-button Mui-disabled\\" tabindex=\\"-1\\" role=\\"button\\" aria-disabled=\\"true\\" style=\\"color: rgb(212, 233, 231);\\"><div class=\\"MuiListItemIcon-root\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path></svg></div><div class=\\"MuiListItemText-root\\"><span class=\\"MuiTypography-root MuiListItemText-primary MuiTypography-body1\\">Create a group</span></div></div></ul></div></div><div class=\\"rightFluid\\" style=\\"padding-top: 0px;\\"><div style=\\"position: relative;\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary\\" tabindex=\\"0\\" type=\\"button\\" style=\\"position: absolute; top: 0px; right: 0px; z-index: 100;\\"><span class=\\"MuiButton-label\\"><svg class=\\"MuiSvgIcon-root buttonLabelIcon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M10 18h4v-2h-4v2zM3 6v2h18V6H3zm3 7h12v-2H6v2z\\"></path></svg>Filters</span><span class=\\"MuiTouchRipple-root\\"></span></button></div><div class=\\"relative\\"><div class=\\"loaderContainer\\"><div class=\\"loader\\"><span class=\\"dot dot_1\\"></span><span class=\\"dot dot_2\\"></span><span class=\\"dot dot_3\\"></span><span class=\\"dot dot_4\\"></span></div></div><div class=\\"hidden\\"><p>No devices found</p><p>No devices have been authorized to connect to the Mender server yet.</p></div><div></div></div></div></div></div>"`;
